<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <arg name = "cam_left_topic"  default = "/kitti/camera_gray_left/image"/>
    <arg name = "cam_right_topic" default = "/kitti/camera_gray_right/image"/>
    <arg name = "imu_topic" default = "/kitti/oxts/imu"/>

    <param name = "frame" type = "string" value = "NED" />

    <rosparam file = "$(find vo)/../../../config/kitti/camera.yaml" />

    <node 
        name   = "vo"
        pkg    = "vo"
        type   = "vo_node"
        args   = "$(arg cam_left_topic) 
                  $(arg cam_right_topic)
                  $(arg imu_topic)"
        output = "screen"
    >
    
    <remap from = "cam_left"  to = "$(arg cam_left_topic)"/>
    <remap from = "cam_right" to = "$(arg cam_right_topic)"/>
    <remap from = "imu_topic" to = "$(arg imu_topic)"/>
    </node>
</launch>
